crates/codegen/runtime/generator/src/parser/parser_definition.rs

```diff
@@ -1,20 +1,20 @@
+use codegen_language_definition::model::{Identifier, VersionSpecifier};
 use inflector::Inflector;
 use proc_macro2::TokenStream;
 use quote::{format_ident, quote};
-use semver::Version;
 
 use crate::parser::grammar::{
-    Labeled, ParserDefinitionNode, ParserDefinitionRef, TriviaParserDefinitionRef, VersionQuality,
-    VersionQualityRange,
+    Labeled, ParserDefinitionNode, ParserDefinitionRef, TriviaParserDefinitionRef,
 };
+use crate::parser::versioned::{Versioned as _, VersionedQuote as _};
 
```

Comment:
I suggest removing the `as _` from import statements, to enable unused warnings when it is no longer needed.

Reply:
That's not true: after removing the usages I still get correct ``unused import: `Versioned as _` `` rustc diagnostic; like I mentioned, this is somewhat special-cased by rustc.

Reply:
> this is somewhat special-cased by rustc

hmmm, this is surprising! thanks for pointing out 🤔 
I wonder what is the benefit of that syntax then, rather than just `use Versioned;`? in this case, the underscore is meaningless to rustc?

Reply:
It exists when you want to import a trait but also use a type with the same name - in this case, using the trait in scope allows you to use the trait methods but you can still use the other data type with a given name.